### PR TITLE
Feature/search employees

### PR DESCRIPTION
**Pull request by asmaaramadan99**
## Brief
- Added Login feature , Search employees , Invite users by email to use skill-matrix.

## Changes
 **Login feature** 
  -  Completes Login action with authentication against database.
  -  Handling first time login for new employees with default password.
  -  First time employee resets default password.

**Invite users by email**
  - An interface for adding emails that invitations will be sent to.
     # issue 
               Configuration issue Emails sent are blocked by gmail.

**Search employees** 
 - Search by employee name.
 - Search by skill name. (default all levels). 
 - Search by skill and level.

### Tests
 **Backend Tests**
- test login page 
- test employee search
- test employee set password

**Front-end test** 
- test for search page

### CODE REVIEW

File: skills-matrix-server/src/main/java/com/cegedim/skillsmatrix/controllers/AddNewUser.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	 * @throws MessagingException
+	 * @throws IOException
+	 */
+	@PostMapping("/send")
+	public void AddUser(@RequestBody List< String > list) throws AddressException, MessagingException, IOException {
+		for(int i=0; i<list.size();i++) {
+			String email = list.get(i);
+			if(service.validateEmail(email)) {
+				System.out.println("After if-condition");
+				System.out.println(email);
```
Comment (hodasamy-cegedim) [inline skills-matrix-server/src/main/java/com/cegedim/skillsmatrix/controllers/AddNewUser.java:48]: please remove the System.out.println from the if condition ,in practical we are not allowed to print this lines except on local environments

File: skills-matrix-server/src/main/java/com/cegedim/skillsmatrix/controllers/CategoriesController.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+@RestController
+@CrossOrigin
+@RequestMapping("/api/categories")
+public class CategoriesController {
+	
+	@Autowired
+	CategoriesService categoriesService;
+	
+	
+	@GetMapping("")
```
Comment (hodasamy-cegedim) [inline skills-matrix-server/src/main/java/com/cegedim/skillsmatrix/controllers/CategoriesController.java:27]: why @GetMapping("") is empty on this class?

File: skills-matrix-server/src/main/java/com/cegedim/skillsmatrix/controllers/JobRolesController.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+@RestController
+@CrossOrigin
+@RequestMapping("/api/roles")
+public class JobRolesController {
+
+	@Autowired
+	private JobRolesService jobRolesService;
+	
+	@GetMapping("")
```
Comment (hodasamy-cegedim) [inline skills-matrix-server/src/main/java/com/cegedim/skillsmatrix/controllers/JobRolesController.java:22]: why @@GetMapping("") is empty

File: skills-matrix-server/src/main/java/com/cegedim/skillsmatrix/controllers/AddNewUser.java (hunk 1 above)
Comment (Abdallah-Youssef) [reply skills-matrix-server/src/main/java/com/cegedim/skillsmatrix/controllers/AddNewUser.java:48]: Noted. I will fix it right away.

Comment (hodasamy-cegedim) [reply skills-matrix-server/src/main/java/com/cegedim/skillsmatrix/controllers/AddNewUser.java]: Great Thanks

**Review by hodasamy-cegedim — CHANGES_REQUESTED** [review]
please follow the comments

**Review by TalaatHarb — APPROVED** [review]

**Comment by Abdallah-Youssef** [comment]
> please follow the comments

Please review the fix in this [commit](https://github.com/TalaatHarb/cgdm-skills-matrix/pull/3/commits/457511d67960dc849c5398e88255bf2e4e91ddd0)

**Review by hodasamy-cegedim — APPROVED** [review]

**Pull request merged**